documentation/latest/intro/concepts.md

```diff
@@ -0,0 +1,145 @@
+  end
+end
+{% endmermaid %}
+
+1. Flotta device-agent(Agent) register to Yggdrasil using Unixsocket and GRPC
+   connection
+2. Yggdrasil allows device-worker to register in the platform.
+3. Agent needs to register to the API, and for that sends a data-message with
+   the registration directive that includes a Certificate Signed Request(CSR).
+4. Yggdrasil forward that message in the configured transport to the Operator API
+5. If registration is successful, it will return a signed certificate based on
+   the CSR
+6. Yggdrasil sends to the agent the response. Agent rewrite the client certs to
+   create new connections with the device certificates.
+7. -10 Each minute, Agent sends heartbeat data to the operator, if the operator
+returns 200 OK nothing happens, if returns 401 for the certificate expired the
+registration process will be started again.
+
+## EdgeDevice
+
+An Edgedevice can be multiple things, from a tiny RPI Zero to a big X86 server.
+For flotta, an Edgedevie is an Operating system where device-agent and Yggdrasil
+are running, and some container workloads can be run by this daemon.
+
+The communication from the edgedevice to the Kubernetes server is not required,
+so all communication is to the Fleet management API. At the same time, there are
+no things like CNI, CSI interfaces for getting it lightweight.
+
+What project-flotta focused is that edgedevices can have one or multiple specs:
+
+- Networks can be flaky, like a satellite, NB-IoT or Lorawan.
+- Power consumption is a high priority; these devices can run on batteries.
+- Devices can be turned off for a long period.
+- Devices are insecure by default.
+- Communication should always be encrypted.
+
+### Yggdrasil
+
+Is a client daemon that establishes a receiving queue for instructions to be
+sent to the system via a broker. For us, the broker is HTTP, and it’s the Fleet
+management API.
+
+### Device Agent
+
+Device Agent is a worker for Yggdrasil, and it's the agent that deploys all
+podman, systemd, logs for all workloads, and deals with the Commmunication to
+the Fleet management API.
+
+## EdgeDevice Signed Request
+
+EdgeDevice Signed Request is a specific [CRD
+type](../operations/crd.html#edgedevicesignedrequest). When AUTO_APPROVAL is set
+to false, when device tries to register, but before it, and Edge Administrator
+needs to approve where the device will land and what kind of configuration it'll
+have.
+
+## DeviceSets
+
+When a platform has thousands of different devices, a configuration group is
+needed to have similar configurations across different devices. A [Device
+Set](../operations/crd.html#edgedeviceset) is one kind of CRD that allows users
+to define this configuration.
+
+Example:
+
+```yaml
+apiVersion: management.project-flotta.io/v1alpha1
+kind: EdgeDeviceSet
+metadata:
+  name: set-1
+spec:
+  heartbeat:
+    periodSeconds: 5
+  metrics:
+    system:
+      interval: 600
+  osInformation:
+    automaticallyUpgrade: true
+```
```

Comment:
Do you plan to add also a section about EdgeWorkload and EdgeDevice? perhaps later to extend also the EdgeConfig :-)

Reply:
This is already on the gsg/running_workloads.md, should be here to?

Reply:
I believe it belongs here since it is main concept of flotta: workloads & configuration.
I just wonder if in terms of components' naming we should go with a generic one and not pin it to the CRD name (which will be detailed in). 
So components names will be:
* Security 
  * EdgeDevice Signed Request
* Workloads
  * EdgeWorkload
* Configuration
  * DeviceSet
  * EdgeConfig

Reply:
But at the same point, we're going to have a guide on operations, around `EdgeWorkload/EdgeDeviceSets /EdgeConfig`

Because is not only a concept, it's bigger than that, and should have a custom entry for each one. IMHO.